vllm/v1/core/sched/scheduler.py

```diff
@@ -330,11 +353,21 @@ def schedule(self) -> SchedulerOutput:
                     new_encoder_budget = encoder_budget
 
                 new_blocks = self.kv_cache_manager.allocate_slots(
-                    request, num_new_tokens, computed_blocks)
+                    request=request,
+                    num_tokens=num_new_tokens,
+                    new_computed_blocks=computed_blocks,
+                    num_external_tokens=num_external_tokens)
```

Comment:
A dumb question: Can we just do this?
```suggestion
                    request=request,
                    num_tokens=num_new_tokens + num_external_tokens,
                    new_computed_blocks=computed_blocks)
```

Reply:
yes, we can actually 😮‍💨